Latest changes from dependabot-core main

AlekhyaYalla opened the pull request:
The latest changes from the dependabot-core contains the fixes for

- tar file path dependency issue ([here](https://github.com/dependabot/dependabot-core/issues/4218))
- creating the .npmrc file with incorrect default registry ([here](https://github.com/dependabot/dependabot-core/issues/4111))
- Optimizations in yarn updater ([here](https://github.com/dependabot/dependabot-core/issues/4288))
- Check & raise 403 TagsCreationForbidden exception in Azure client. ([here](https://github.com/dependabot/dependabot-core/pull/4352))

xlgmokha committed:
Merge pull request #4179 from xlgmokha/pip-slo

fix: Parse multiple requirements from a poetry dependency

xlgmokha committed:
Merge pull request #4185 from dependabot/build-cron

chore: run builds on a regular basis to detect anomolies

jurre committed:
Merge pull request #4206 from dependabot/jurre/python-incompatible-versions

Python: Raise resolvability error with explanation when update is not possible

xlgmokha committed:
v0.162.0

xlgmokha committed:
Merge pull request #4210 from xlgmokha/v0.162.0-release-notes

v0.162.0

Nishnha committed:
Merge pull request #4191 from declan-fitzpatrick/main

handle terraform module versions with a 'v' prefix

xlgmokha committed:
Merge pull request #4207 from pauloromeira/pyenv-2.0.6

Python: Upgrade pyenv to 2.0.6

nirev committed:
handle support files with dir macro in umbrellas

brrygrdn committed:
bin/dry-run.rb exits outside a developer shell

brrygrdn committed:
Update README

brrygrdn committed:
Prefer checking user over touching a file

brrygrdn committed:
Merge pull request #4215 from dependabot/brrygrdn/dry-run-requires-dev-shell

bin/dry-run.rb requires a development container to run

zbynek committed:
Support basic Kotlin apply, add tests

jurre committed:
Escape paths passed to VendorUpdater

Paths in packages could be constructed to perform command, when not
properly escaped those could be executed.

```
(byebug) `file -b --mime-encoding t&&curl$IFS@0.0.0.0&&.go`
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (7) Failed to connect to 0.0.0.0 port 80: Connection refused
"cannot open `t' (No such file or directory)\n"
```

```
(byebug) `#{Dependabot::SharedHelpers.escape_command("file -b --mime-encoding t&&curl$IFS@0.0.0.0&&.go")}`
"cannot open `t&&curl$IFS@0.0.0.0&&.go' (No such file or directory)\n"
```

jurre committed:
Merge pull request #4221 from dependabot/jurre/escape-vendor-updater-cmd

Escape paths passed to VendorUpdater

mctofu committed:
Bump golang from 1.17 to 1.17.1

jurre committed:
Merge pull request #4155 from zbynek/gradle-includes

Improved support `apply from` in gradle files

dependabot[bot] committed:
build(deps): bump github.com/dependabot/gomodules-extracted

Bumps [github.com/dependabot/gomodules-extracted](https://github.com/dependabot/gomodules-extracted) from 1.4.1 to 1.4.2.
- [Release notes](https://github.com/dependabot/gomodules-extracted/releases)
- [Commits](https://github.com/dependabot/gomodules-extracted/compare/v1.4.1...v1.4.2)

---
updated-dependencies:
- dependency-name: github.com/dependabot/gomodules-extracted
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

mctofu committed:
Merge pull request #4231 from dependabot/mctofu/go1.17.1

Bump golang from 1.17 to 1.17.1

mctofu committed:
Merge pull request #4235 from dependabot/dependabot/go_modules/go_modules/helpers/github.com/dependabot/gomodules-extracted-1.4.2

build(deps): bump github.com/dependabot/gomodules-extracted from 1.4.1 to 1.4.2 in /go_modules/helpers

meladRaouf committed:
fix  `labels': Unsupported provider bitbucket

Bitbucket doesn't yet supports PR labels.
dependabot throught this issue whenever I try to use it with bitbucket 
labeler.rb:241:in `labels': Unsupported provider bitbucket (RuntimeError)

zbynek committed:
Fix minor typos in changelog

xlgmokha committed:
Merge pull request #4237 from zbynek/patch-1

Fix minor typos in changelog

mctofu committed:
v0.162.1

Commit:
Handle .tar path dependency

mctofu committed:
Merge pull request #4245 from dependabot/v0.162.1-release-notes

v0.162.1

baseballlover723 committed:
update Elixir from 1.12.2 -> 1.12.3

zbynek committed:
Support Gradle files with no top level build.gradle file

zbynek committed:
Treat tokens after underscore as numeric if possible

zbynek committed:
Simplify string processing

jurre committed:
Merge pull request #4257 from zbynek/underscore-numeric-token

Treat tokens after underscore as numeric if possible

jerbob92 committed:
Ignore replaced dependencies

Nishnha committed:
Merge pull request #4140 from jerbob92/feature/go-mod-replace-support

Ignore replaced dependencies in go.mod

Nishnha committed:
v0.162.2

zbynek committed:
Add support for gradlePluginPortal()

dependabot[bot] committed:
build(deps-dev): bump jest in /npm_and_yarn/helpers

Bumps [jest](https://github.com/facebook/jest) from 27.0.6 to 27.2.4.
- [Release notes](https://github.com/facebook/jest/releases)
- [Changelog](https://github.com/facebook/jest/blob/main/CHANGELOG.md)
- [Commits](https://github.com/facebook/jest/compare/v27.0.6...v27.2.4)

---
updated-dependencies:
- dependency-name: jest
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Commit:
Fix indentation

baseballlover723 committed:
update erlang solutions from 1.0 to 2.0

dependabot[bot] committed:
build(deps): bump composer/composer in /composer/helpers/v1

Bumps [composer/composer](https://github.com/composer/composer) from 1.10.22 to 1.10.23.
- [Release notes](https://github.com/composer/composer/releases)
- [Changelog](https://github.com/composer/composer/blob/master/CHANGELOG.md)
- [Commits](https://github.com/composer/composer/compare/1.10.22...1.10.23)

---
updated-dependencies:
- dependency-name: composer/composer
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump @npmcli/arborist in /npm_and_yarn/helpers

Bumps [@npmcli/arborist](https://github.com/npm/arborist) from 2.8.0 to 3.0.0.
- [Release notes](https://github.com/npm/arborist/releases)
- [Changelog](https://github.com/npm/arborist/blob/main/CHANGELOG.md)
- [Commits](https://github.com/npm/arborist/compare/v2.8.0...v3.0.0)

---
updated-dependencies:
- dependency-name: "@npmcli/arborist"
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump friendsofphp/php-cs-fixer in /composer/helpers/v2

Bumps [friendsofphp/php-cs-fixer](https://github.com/FriendsOfPHP/PHP-CS-Fixer) from 3.0.0 to 3.2.1.
- [Release notes](https://github.com/FriendsOfPHP/PHP-CS-Fixer/releases)
- [Changelog](https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/v3.2.1/CHANGELOG.md)
- [Commits](https://github.com/FriendsOfPHP/PHP-CS-Fixer/compare/v3.0.0...v3.2.1)

---
updated-dependencies:
- dependency-name: friendsofphp/php-cs-fixer
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

andrewbredow committed:
Add issue template config with Dependabot contact links

jurre committed:
Merge pull request #4246 from AlekhyaYalla/alekhyayalla/handle_tar_path_dependencies

Handle .tar path dependency

jurre committed:
Fix elm tests

The version resolver fetches the latest version from the public
registry, and since we shell out to `elm` for this, it's currently hard
to stub out.

Our test fixture did not have the latest version yet and would cause our
tests to fail.

Ideally we'd rely on both the version_checker and resolver using
the same source that we stub, but unfortunately with the way our tests
are set up makes that hard. Fortunately these don't change often, so I
suggest we live with the pain for now.

jurre committed:
Merge pull request #4294 from dependabot/jurre/fix-elm-tests

Fix elm tests

jurre committed:
Fix Poetry unreachable git deps error

Since https://github.com/python-poetry/poetry-core/pull/202 poetry now
uses a slightly different (safer) git command, and this caused the
regex matching we do on the error output to now fail.

This fixes up the regex, and just to be safe ensures it'll keep working
with the old version as well.

The test that was failing was:
`python/spec/dependabot/python/update_checker/poetry_version_resolver_spec.rb:179`

jurre committed:
Merge pull request #4289 from andrewbredow/patch-1

Add issue template config with Dependabot contact links

jurre committed:
Merge pull request #4295 from dependabot/jurre/fix-python-tests

Fix Poetry unreachable git deps error

jurre committed:
Merge pull request #4268 from zbynek/gradle-plugin-portal

Add support for gradlePluginPortal()

jurre committed:
Merge pull request #4271 from dependabot/dependabot/npm_and_yarn/npm_and_yarn/helpers/jest-27.2.4

build(deps-dev): bump jest from 27.0.6 to 27.2.4 in /npm_and_yarn/helpers

dependabot[bot] committed:
build(deps-dev): bump prettier in /npm_and_yarn/helpers

Bumps [prettier](https://github.com/prettier/prettier) from 2.3.2 to 2.4.1.
- [Release notes](https://github.com/prettier/prettier/releases)
- [Changelog](https://github.com/prettier/prettier/blob/main/CHANGELOG.md)
- [Commits](https://github.com/prettier/prettier/compare/2.3.2...2.4.1)

---
updated-dependencies:
- dependency-name: prettier
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

jurre committed:
Gradle: Prefer method call over instance variable access in file fetcher

dependabot[bot] committed:
build(deps): bump poetry from 1.1.7 to 1.1.11 in /python/helpers

Bumps [poetry](https://github.com/python-poetry/poetry) from 1.1.7 to 1.1.11.
- [Release notes](https://github.com/python-poetry/poetry/releases)
- [Changelog](https://github.com/python-poetry/poetry/blob/1.1.11/CHANGELOG.md)
- [Commits](https://github.com/python-poetry/poetry/compare/1.1.7...1.1.11)

---
updated-dependencies:
- dependency-name: poetry
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

jurre committed:
Merge pull request #4283 from dependabot/dependabot/composer/composer/helpers/v1/composer/composer-1.10.23

build(deps): bump composer/composer from 1.10.22 to 1.10.23 in /composer/helpers/v1

jurre committed:
Merge pull request #4284 from dependabot/dependabot/npm_and_yarn/npm_and_yarn/helpers/npmcli/arborist-3.0.0

build(deps): bump @npmcli/arborist from 2.8.0 to 3.0.0 in /npm_and_yarn/helpers

dependabot[bot] committed:
build(deps): bump composer/composer in /composer/helpers/v2

Bumps [composer/composer](https://github.com/composer/composer) from 2.1.3 to 2.1.9.
- [Release notes](https://github.com/composer/composer/releases)
- [Changelog](https://github.com/composer/composer/blob/master/CHANGELOG.md)
- [Commits](https://github.com/composer/composer/compare/2.1.3...2.1.9)

---
updated-dependencies:
- dependency-name: composer/composer
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump pip-tools from 6.2.0 to 6.3.0 in /python/helpers

Bumps [pip-tools](https://github.com/jazzband/pip-tools) from 6.2.0 to 6.3.0.
- [Release notes](https://github.com/jazzband/pip-tools/releases)
- [Changelog](https://github.com/jazzband/pip-tools/blob/master/CHANGELOG.md)
- [Commits](https://github.com/jazzband/pip-tools/compare/6.2.0...6.3.0)

---
updated-dependencies:
- dependency-name: pip-tools
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

jurre committed:
Merge pull request #4251 from dependabot/dependabot/pip/python/helpers/pip-tools-6.3.0

build(deps): bump pip-tools from 6.2.0 to 6.3.0 in /python/helpers

jurre committed:
Merge pull request #4281 from dependabot/dependabot/pip/python/helpers/poetry-1.1.11

build(deps): bump poetry from 1.1.7 to 1.1.11 in /python/helpers

jurre committed:
Merge pull request #4256 from zbynek/settings-only

Support Gradle files with no top level build.gradle file

dependabot[bot] committed:
build(deps): bump wheel from 0.36.2 to 0.37.0 in /python/helpers

Bumps [wheel](https://github.com/pypa/wheel) from 0.36.2 to 0.37.0.
- [Release notes](https://github.com/pypa/wheel/releases)
- [Changelog](https://github.com/pypa/wheel/blob/master/docs/news.rst)
- [Commits](https://github.com/pypa/wheel/compare/0.36.2...0.37.0)

---
updated-dependencies:
- dependency-name: wheel
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

jurre committed:
Merge pull request #4286 from dependabot/dependabot/composer/composer/helpers/v2/composer/composer-2.1.9

build(deps): bump composer/composer from 2.1.3 to 2.1.9 in /composer/helpers/v2

jurre committed:
Merge pull request #4236 from meladRaouf/patch-1

fix  `labels': Unsupported provider bitbucket

dependabot[bot] committed:
build(deps-dev): bump phpstan/phpstan in /composer/helpers/v1

Bumps [phpstan/phpstan](https://github.com/phpstan/phpstan) from 0.12.93 to 0.12.99.
- [Release notes](https://github.com/phpstan/phpstan/releases)
- [Commits](https://github.com/phpstan/phpstan/compare/0.12.93...0.12.99)

---
updated-dependencies:
- dependency-name: phpstan/phpstan
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

jurre committed:
Merge pull request #4220 from dependabot/dependabot/composer/composer/helpers/v1/phpstan/phpstan-0.12.99

jurre committed:
Merge pull request #4285 from dependabot/dependabot/composer/composer/helpers/v2/friendsofphp/php-cs-fixer-3.2.1

dependabot[bot] committed:
build(deps-dev): bump friendsofphp/php-cs-fixer in /composer/helpers/v1

Bumps [friendsofphp/php-cs-fixer](https://github.com/FriendsOfPHP/PHP-CS-Fixer) from 2.19.0 to 2.19.2.
- [Release notes](https://github.com/FriendsOfPHP/PHP-CS-Fixer/releases)
- [Changelog](https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/3.0/CHANGELOG.md)
- [Commits](https://github.com/FriendsOfPHP/PHP-CS-Fixer/compare/v2.19.0...v2.19.2)

---
updated-dependencies:
- dependency-name: friendsofphp/php-cs-fixer
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

jurre committed:
Merge pull request #4266 from dependabot/v0.162.2-release-notes

v0.162.2

jurre committed:
v0.163.0

jurre committed:
Merge pull request #4296 from dependabot/v0.163.0-release-notes

v0.163.0

jurre committed:
Merge pull request #4233 from dependabot/dependabot/npm_and_yarn/npm_and_yarn/helpers/prettier-2.4.1

build(deps-dev): bump prettier from 2.3.2 to 2.4.1 in /npm_and_yarn/helpers

jurre committed:
Merge pull request #4128 from dependabot/dependabot/pip/python/helpers/wheel-0.37.0

build(deps): bump wheel from 0.36.2 to 0.37.0 in /python/helpers

jurre committed:
Merge pull request #4152 from dependabot/dependabot/composer/composer/helpers/v1/friendsofphp/php-cs-fixer-2.19.2

build(deps-dev): bump friendsofphp/php-cs-fixer from 2.19.0 to 2.19.2 in /composer/helpers/v1

jurre committed:
Merge pull request #4199 from Marcono1234/patch-3

Clarify that Remote Containers Clone Repository commands are not supported

nirev committed:
remove trailing whitespaces

Marcono1234 committed:
Recommend `git reset` when cloning repository on Windows fails

`git restore --source=HEAD :/` which is currently suggested by Git does not work,
it is unable to restore the files, see https://github.com/git-for-windows/git/issues/3411.

dependabot[bot] committed:
build(deps): bump pip-tools from 6.3.0 to 6.3.1 in /python/helpers

Bumps [pip-tools](https://github.com/jazzband/pip-tools) from 6.3.0 to 6.3.1.
- [Release notes](https://github.com/jazzband/pip-tools/releases)
- [Changelog](https://github.com/jazzband/pip-tools/blob/master/CHANGELOG.md)
- [Commits](https://github.com/jazzband/pip-tools/compare/6.3.0...6.3.1)

---
updated-dependencies:
- dependency-name: pip-tools
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump jest in /npm_and_yarn/helpers

Bumps [jest](https://github.com/facebook/jest) from 27.2.4 to 27.2.5.
- [Release notes](https://github.com/facebook/jest/releases)
- [Changelog](https://github.com/facebook/jest/blob/main/CHANGELOG.md)
- [Commits](https://github.com/facebook/jest/compare/v27.2.4...v27.2.5)

---
updated-dependencies:
- dependency-name: jest
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

jurre committed:
Merge pull request #4302 from dependabot/dependabot/pip/python/helpers/pip-tools-6.3.1

build(deps): bump pip-tools from 6.3.0 to 6.3.1 in /python/helpers

dependabot[bot] committed:
build(deps): bump pip from 21.1.3 to 21.2.4 in /python/helpers

Bumps [pip](https://github.com/pypa/pip) from 21.1.3 to 21.2.4.
- [Release notes](https://github.com/pypa/pip/releases)
- [Changelog](https://github.com/pypa/pip/blob/main/NEWS.rst)
- [Commits](https://github.com/pypa/pip/compare/21.1.3...21.2.4)

---
updated-dependencies:
- dependency-name: pip
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump flake8 from 3.9.2 to 4.0.0 in /python/helpers

Bumps [flake8](https://github.com/pycqa/flake8) from 3.9.2 to 4.0.0.
- [Release notes](https://github.com/pycqa/flake8/releases)
- [Commits](https://github.com/pycqa/flake8/compare/3.9.2...4.0.0)

---
updated-dependencies:
- dependency-name: flake8
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

jurre committed:
Merge pull request #4304 from dependabot/dependabot/npm_and_yarn/npm_and_yarn/helpers/jest-27.2.5

build(deps-dev): bump jest from 27.2.4 to 27.2.5 in /npm_and_yarn/helpers

dependabot[bot] committed:
build(deps-dev): bump eslint in /npm_and_yarn/helpers

Bumps [eslint](https://github.com/eslint/eslint) from 7.32.0 to 8.0.0.
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/master/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v7.32.0...v8.0.0)

---
updated-dependencies:
- dependency-name: eslint
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump golang.org/x/mod in /go_modules/helpers

Bumps [golang.org/x/mod](https://github.com/golang/mod) from 0.5.0 to 0.5.1.
- [Release notes](https://github.com/golang/mod/releases)
- [Commits](https://github.com/golang/mod/compare/v0.5.0...v0.5.1)

---
updated-dependencies:
- dependency-name: golang.org/x/mod
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

jurre committed:
Merge pull request #4303 from dependabot/dependabot/pip/python/helpers/flake8-4.0.0

build(deps): bump flake8 from 3.9.2 to 4.0.0 in /python/helpers

jurre committed:
Merge pull request #4301 from dependabot/dependabot/npm_and_yarn/npm_and_yarn/helpers/eslint-8.0.0

build(deps-dev): bump eslint from 7.32.0 to 8.0.0 in /npm_and_yarn/helpers

jurre committed:
Merge pull request #4136 from dependabot/dependabot/pip/python/helpers/pip-21.2.4

build(deps): bump pip from 21.1.3 to 21.2.4 in /python/helpers

jurre committed:
Merge pull request #4254 from dependabot/dependabot/go_modules/go_modules/helpers/golang.org/x/mod-0.5.1

build(deps): bump golang.org/x/mod from 0.5.0 to 0.5.1 in /go_modules/helpers

jurre committed:
Merge pull request #4198 from Marcono1234/patch-1

Describe workaround for cloning with long file names on Windows

tomaspinho committed:
Upgrade OTP to latest minor: 23.3.4.5

jurre committed:
Merge pull request #4213 from nirev/hex-replace-dir-macro

hex: handle support files with macro in umbrellas

jurre committed:
Merge pull request #4306 from tomaspinho/main

Upgrade OTP to latest minor: 23.3.4.5

Parnassius committed:
Python: Upgrade pyenv to 2.1.0

This also adds Python 3.10.0, 3.7.12, 3.6.15

jurre committed:
Merge pull request #4307 from Parnassius/pyenv-2-1-0

Python: Upgrade pyenv to 2.1.0

jurre committed:
Bump Terraform from 1.0.6 to 1.0.8

https://github.com/hashicorp/terraform/blob/v1.0/CHANGELOG.md#108-september-29-2021

dependabot[bot] committed:
build(deps): bump pip-tools from 6.3.1 to 6.4.0 in /python/helpers

Bumps [pip-tools](https://github.com/jazzband/pip-tools) from 6.3.1 to 6.4.0.
- [Release notes](https://github.com/jazzband/pip-tools/releases)
- [Changelog](https://github.com/jazzband/pip-tools/blob/master/CHANGELOG.md)
- [Commits](https://github.com/jazzband/pip-tools/compare/6.3.1...6.4.0)

---
updated-dependencies:
- dependency-name: pip-tools
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

jurre committed:
Merge pull request #4312 from dependabot/dependabot/pip/python/helpers/pip-tools-6.4.0

build(deps): bump pip-tools from 6.3.1 to 6.4.0 in /python/helpers

dependabot[bot] committed:
build(deps): bump flake8 from 4.0.0 to 4.0.1 in /python/helpers

Bumps [flake8](https://github.com/pycqa/flake8) from 4.0.0 to 4.0.1.
- [Release notes](https://github.com/pycqa/flake8/releases)
- [Commits](https://github.com/pycqa/flake8/compare/4.0.0...4.0.1)

---
updated-dependencies:
- dependency-name: flake8
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

jurre committed:
Merge pull request #4310 from dependabot/jurre/update-terraform

Bump Terraform from 1.0.6 to 1.0.8

jurre committed:
Merge pull request #4308 from dependabot/dependabot/pip/python/helpers/flake8-4.0.1

build(deps): bump flake8 from 4.0.0 to 4.0.1 in /python/helpers

jurre committed:
Merge pull request #4250 from baseballlover723/pr/update_elixir

update Elixir from 1.12.2 -> 1.12.3

jurre committed:
v0.163.1

jurre committed:
Merge pull request #4316 from dependabot/v0.163.1-release-notes

v0.163.1

Nishnha committed:
Add license to image and gemspec

The dependabot-* gems on rubygems currently have the license set as 'nonstandard' and do not include a license file.

Nishnha committed:
Add license to gem build directory

Nishnha committed:
Set license back to Nonstandard

Rubygems only recognizes licenses on the spdx license list, and the Prosperity Public License 2.0.0 is not one of them.

andrcuns committed:
Allow passing target_project_id to Gitlab pr creator and updater

Nishnha committed:
Revert "Add license to gem build directory"

This reverts commit 05aa6ee50fd752f8d3c7c9fae1e2af86489c2f90.

andrcuns committed:
Update specs

Validate target_project_id passed correctly

Fix request expectation

andrcuns committed:
Move target_project_id to provider_metadata

jurre committed:
Merge pull request #4324 from andrcuns/gitlab-add-fork-support

Gitlab: add support for creating merge requests for forks

Nishnha committed:
Merge pull request #4317 from dependabot/nishnha/add-license

Add license to image and gemspec

jurre committed:
Treat GHES hosted sources as github sources

When a dependency is hosted on GHES, previously it was not treated as a
GitHub source, meaning that we would not check for releases/changelogs
etc when requesting Metadata for the PR.

This fixes that, by first parsing the URL, and then making a request to
`<host>/status`, and checking for a `X-GitHub-Request-Id` header, which
we return from GitHub Enterprise Server.

phillipuniverse committed:
Do not freeze file-based Poetry dependency version

Fixes #4333

phillipuniverse committed:
Polish

phillipuniverse committed:
Code formatting

phillipuniverse committed:
Remove unnecessary quotes

Co-authored-by: Jurre <jurre@github.com>

phillipuniverse committed:
Fix typo

brrygrdn committed:
Only check auth for github.com

brrygrdn committed:
Revert "v0.163.1"

This reverts commit 2103fbb41d1017ff1939cc9789ef9dab216b37c4.

brrygrdn committed:
v0.164.0

brrygrdn committed:
Merge pull request #4348 from dependabot/v0.164.0-release-notes

v0.164.0 release notes

Commit:
Add TagsCreationForbidden Exception to Azure Client

Commit:
Make labeler optional in Azure

Commit:
Add labels_required param to PR creator

Commit:
Fix indentation

jurre committed:
Merge pull request #4335 from dependabot/jurre/ghes-host

Treat GHES hosted sources as github sources

brrygrdn committed:
Ensure we cleanup tmp directories after use

brrygrdn committed:
Merge pull request #4356 from dependabot/brrygrdn/cleanup-tmp-dirs

Ensure we cleanup tmp directories after use

brrygrdn committed:
Merge pull request #4347 from dependabot/fix-bump-version

Only check auth for github.com when running `bump-version`

brrygrdn committed:
v0.164.1

brrygrdn committed:
Merge pull request #4360 from dependabot/v0.164.1-release-notes

v0.164.1

xlgmokha committed:
feat: specify timeouts per spawned process

xlgmokha committed:
style: fix linter errors

xlgmokha committed:
test: update glass assertions

xlgmokha committed:
test: ensure the proper command is generated

xlgmokha committed:
feat: trap SIGHUB and flush error to stdout

xlgmokha committed:
refactor: collapse multi line conditional

xlgmokha committed:
style: fix linter errors

xlgmokha committed:
refactor: extract class to build shell command

xlgmokha committed:
style: fix linter errors

xlgmokha committed:
fix: ensure min of 1 minute and max of 30 minutes per op

xlgmokha committed:
refactor: make timeout_seconds a private method

anatawa12 committed:
[Gradle] add settings_file to fetched_files

anatawa12 committed:
[Gradle] add settings files to SUPPORTED_BUILD_FILE_NAMES

anatawa12 committed:
[Gradle] update tests

xlgmokha committed:
refactor: use Comparable#clamp

anatawa12 committed:
[Gradle] add tests for FileParser

anatawa12 committed:
[Gradle] fix implementation

xlgmokha committed:
Merge pull request #4362 from dependabot/xlg/1918-timeout-per-op

Add timeout per operation

xlgmokha committed:
v0.165.0

xlgmokha committed:
Merge pull request #4387 from dependabot/v0.165.0-release-notes

v0.165.0

jeffwidman committed:
Bump to go 1.17.3

There were some minor fixes to the `go` command in 1.17.2/1.17.3... 

I don't think any of them directly affect dependabot, but it
doesn't hurt to bump this and makes it so the next person has a smaller
diff to look at when they consider updating.

mctofu committed:
Move composer-not-found fixture from decommissioned dependabot.com

mctofu committed:
Merge pull request #4399 from dependabot/mctofu/composer-not-found-fixture

Move composer-not-found fixture from decommissioned dependabot.com

mctofu committed:
Merge pull request #4393 from jeffwidman/patch-1

Bump to go 1.17.3

jurre committed:
Ignore errors from Source enterprise check and ignore known failures

We check if a potential Source is GitHub enterprise by making a request
to a `/status` endpoint against the root URL and checking some headers.

We've observed this check failing in some cases when the source is not
enterprise, and we get rate limited, or otherwise the request fails with
an error.

In this case we do not want to block creating a PR, but instead we
should assume the source is not Enterprise.

This also adds a list of known hosts that we come across often that
definitely are not GitHub Enterprise instances, and we ignore those and
don't bother making a request to them.

Co-Authored-By: Barry Gordon <brrygrdn@github.com>

jurre committed:
Explicitly ignore metadata detection for fuchsia.googlesource.com

We've observed some failures when trying to establish if this might be
an GHES host, since we can be confident it's not, let's not bother
making a request to check.

jurre committed:
Merge pull request #4401 from dependabot/jurre-brrygrdn/safer-enterprise-checks

Ignore errors from Source enterprise check and ignore known failures

jurre committed:
v0.166.0

jurre committed:
Merge pull request #4403 from dependabot/v0.166.0-release-notes

v0.166.0

jurre committed:
Merge pull request #4402 from dependabot/jurre/ignore-googlesource-metadata

Explicitly ignore metadata detection for fuchsia.googlesource.com

Zastai committed:
Apply suggested code tweaks

Zastai committed:
Fix variable references

Zastai committed:
Merge branch 'main' into msbuild-sdk-references

Zastai committed:
Refactor code to reduce complexity

This also reverts a previous incorrect change.
With this, rubocop is happy and the tests still pass.

jurre committed:
Improve error handling when `terraform init` fails

We attempt to run `terraform init` once, but when this fails, we still
end up with an unhandled error. Given that there is not much we can do
at this point, communicate that to the user instead, and treat it as a
resolvability error.

At the same time, when `terraform init` fails with an error other than
a private source error, there is no point trying further so we should
raise a resolvability error as well.

jurre committed:
Rescue terraform registry connection errors

jurre committed:
Merge pull request #4405 from dependabot/jurre/improve-tf-error-handling

Terraform: Improve file updater and registry request error handling

jurre committed:
v0.166.1

jurre committed:
Merge pull request #4406 from dependabot/v0.166.1-release-notes

v0.166.1

Commit:
Remove labels_required param

Commit:
Run YarnUpdate only once for a version requirement

Commit:
Merge remote-tracking branch 'upstream/main' into test_failures

Commit:
Fix indent

Zastai committed:
Adjust indentation to please rubocop

xlgmokha committed:
fix: remove fixed error message check

A new version of [pandoc-ruby](https://rubygems.org/gems/pandoc-ruby/versions/2.1.5) was recently released
that changes the behaviour of how the `pandoc` executable is invoked.
[src](https://github.com/xwmx/pandoc-ruby/compare/2.1.4...2.1.5).

/cc https://github.com/dependabot/dependabot-core/pull/2849

xlgmokha committed:
style: fix linter errors

xlgmokha committed:
Merge pull request #4412 from dependabot/fix-pandoc

fix: remove fixed error message check

Zastai committed:
Merge branch 'main' into msbuild-sdk-references

Commit:
Fix

Commit:
Merge remote-tracking branch 'upstream/main' into test_failures

xlgmokha committed:
Merge pull request #2849 from Zastai/msbuild-sdk-references

dependabot[bot] committed:
build(deps-dev): bump friendsofphp/php-cs-fixer in /composer/helpers/v2

Bumps [friendsofphp/php-cs-fixer](https://github.com/FriendsOfPHP/PHP-CS-Fixer) from 3.2.1 to 3.3.2.
- [Release notes](https://github.com/FriendsOfPHP/PHP-CS-Fixer/releases)
- [Changelog](https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/v3.3.2/CHANGELOG.md)
- [Commits](https://github.com/FriendsOfPHP/PHP-CS-Fixer/compare/v3.2.1...v3.3.2)

---
updated-dependencies:
- dependency-name: friendsofphp/php-cs-fixer
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump friendsofphp/php-cs-fixer in /composer/helpers/v1

Bumps [friendsofphp/php-cs-fixer](https://github.com/FriendsOfPHP/PHP-CS-Fixer) from 2.19.2 to 2.19.3.
- [Release notes](https://github.com/FriendsOfPHP/PHP-CS-Fixer/releases)
- [Changelog](https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/master/CHANGELOG.md)
- [Commits](https://github.com/FriendsOfPHP/PHP-CS-Fixer/compare/v2.19.2...v2.19.3)

---
updated-dependencies:
- dependency-name: friendsofphp/php-cs-fixer
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump pipenv from 2021.5.29 to 2021.11.15 in /python/helpers

Bumps [pipenv](https://github.com/pypa/pipenv) from 2021.5.29 to 2021.11.15.
- [Release notes](https://github.com/pypa/pipenv/releases)
- [Changelog](https://github.com/pypa/pipenv/blob/main/CHANGELOG.rst)
- [Commits](https://github.com/pypa/pipenv/compare/v2021.5.29...v2021.11.15)

---
updated-dependencies:
- dependency-name: pipenv
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

jurre committed:
Maven: Correctly handle nested declarations

We've observed errors around pom files that have a nested `plugin`
section inside an outer plugin declaration.

Previously we used a regex to recursively scan the XML document, which
caused the inner `plugin` declarations closing tag to match the regex,
resulting in an incomplete XML section.

To demonstrate, given the following XML:

```xml
<plugins>
  <plugin>
    <configuration>
      <jvmTarget>11</jvmTarget>
      <compilerPlugins>
        <plugin>spring</plugin>
      </compilerPlugins>
    </configuration>
    <groupId>org.jetbrains.kotlin</groupId>
    <artifactId>kotlin-maven-plugin</artifactId>
    <version>${kotlin.version}</version>
  </plugin>
</plugin>
```

The `<plugin>spring</plugin>` declaration would cause a regex match,
resulting in the required information (the `version` in this case) to be
omitted from the XML snippet.

This is resolved by using Nokogiri to traverse the XML instead of using
regular expressions, and selecting the nodes by name.

Co-authored-by: Landon Grindheim <landongrindheim@github.com>

jurre committed:
Merge pull request #4417 from dependabot/jurre-landongrindheim/fix-maven-nested-plugins

Maven: Correctly handle nested declarations

landongrindheim committed:
v0.167.0

landongrindheim committed:
Merge pull request #4418 from dependabot/v0.167.0-release-notes

v0.167.0

brrygrdn committed:
Remove the dependabot migration issue template

The Dependabot Preview service was shut down on the 3rd of August 2021,
so our path to migration is now closed and the service has been wrapped
up.

This template format is no longer required, any issues for lapsed users
returning to the integrated Dependabot service should just file a standard
bug report.

GiriB commented:
In this PR I'm seeing commits from August. Didn't we just pull sometime last month? Why are so older commits part of this PR?

phillipuniverse committed:
Special case URL dependencies, add better positive assertions for file and directory dependencies

anatawa12 committed:
Merge branch 'dependabot:main' into settings-gradle-kts

phillipuniverse committed:
Rename expected files in tests

phillipuniverse committed:
Rename file back

jurre committed:
Update some pipenv error handling to match latest version

brrygrdn committed:
Merge pull request #4422 from dependabot/brrygrdn/remove-migration-issue-tmpl

Remove the dependabot migration issue template

mctofu committed:
Switch to go command to find available module versions

mctofu committed:
Handle new error message for bad module paths

go list returns a different error message:
go list -m: malformed module path "pkg-errors": missing dot in first path element

mctofu committed:
Enable retracted module test

go list -m properly handles retractions so this test is now passing

mctofu committed:
Bump latest go-modules-lib to v3. v2 was invalid.

Prior error:
go list -m: loading module retractions for github.com/dependabot-fixtures/go-modules-lib/v2@v2.0.0: version "v2.0.0" invalid: go.mod has non-.../v2 module path "github.com/dependabot-fixtures/go-modules-lib" (and .../v2/go.mod does not exist) at revision v2.0.0

mctofu committed:
Handle invalid major version errors

dependabot[bot] committed:
build(deps): bump @npmcli/arborist in /npm_and_yarn/helpers

Bumps [@npmcli/arborist](https://github.com/npm/arborist) from 3.0.0 to 4.0.5.
- [Release notes](https://github.com/npm/arborist/releases)
- [Changelog](https://github.com/npm/arborist/blob/main/CHANGELOG.md)
- [Commits](https://github.com/npm/arborist/compare/v3.0.0...v4.0.5)

---
updated-dependencies:
- dependency-name: "@npmcli/arborist"
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump phpstan/phpstan in /composer/helpers/v1

Bumps [phpstan/phpstan](https://github.com/phpstan/phpstan) from 0.12.99 to 1.2.0.
- [Release notes](https://github.com/phpstan/phpstan/releases)
- [Changelog](https://github.com/phpstan/phpstan/blob/master/CHANGELOG.md)
- [Commits](https://github.com/phpstan/phpstan/compare/0.12.99...1.2.0)

---
updated-dependencies:
- dependency-name: phpstan/phpstan
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump phpstan/phpstan in /composer/helpers/v2

Bumps [phpstan/phpstan](https://github.com/phpstan/phpstan) from 0.12.93 to 1.2.0.
- [Release notes](https://github.com/phpstan/phpstan/releases)
- [Changelog](https://github.com/phpstan/phpstan/blob/master/CHANGELOG.md)
- [Commits](https://github.com/phpstan/phpstan/compare/0.12.93...1.2.0)

---
updated-dependencies:
- dependency-name: phpstan/phpstan
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

andrcuns committed:
Add support for custom commit message trailer

jurre committed:
Merge pull request #4374 from anatawa12/settings-gradle-kts

Support updating dependencies on settings file

andrcuns committed:
Pass custom git trailers as hash object

landongrindheim committed:
Remove reliance on `PandocRuby`

PandocRuby has been used to convert RestructuredText (rst), a
markdown-like format widely used in the Python ecosystem, to markdown.
We recently noticed new errors surfacing around Pandoc and started to
investigate.  This led to the discovery that Pandoc was not installed in
the Docker container GitHub is using to run Dependabot against
repositories.

I'm opting to remove this dependency as PandocRuby is effectively
unused.

Note: There is the possibility that some users rely on this
functionality. As has been noted in a recent PR-review, non-Docker usage
of dependabot-core is poorly supported, so this seems unlikely.

jurre committed:
Merge pull request #4432 from andrcuns/commit-message-trailer

Add support for custom commit message trailers

jurre committed:
Merge pull request #4429 from dependabot/dependabot/npm_and_yarn/npm_and_yarn/helpers/npmcli/arborist-4.0.5

build(deps): bump @npmcli/arborist from 3.0.0 to 4.0.5 in /npm_and_yarn/helpers

landongrindheim committed:
Update tests to reflect our not converting rst's

We stopped relying on `PandocRuby` in a previous commit. This test may
be able to go away, but first I'd like to prove that it's no longer
needed.

mctofu committed:
Allow providing env to SharedHelpers.run_shell_command

jurre commented:
Seems like a rebase/merge gone wrong. Might be good to reset to main and cherry-pick the relevant changes

mctofu committed:
Remove unused go_modules updatechecker native helper

landongrindheim committed:
Merge pull request #4433 from dependabot/remove-reliance-on-pandoc-ruby

Remove reliance on `PandocRuby`

AlekhyaYalla committed:
Fix linter error

Co-authored-by: Jurre <jurre@github.com>

Commit:
Verify for just message code instead of text

AlekhyaYalla committed:
Fix linter issue

Co-authored-by: Jurre <jurre@github.com>

dependabot[bot] committed:
build(deps-dev): bump eslint in /npm_and_yarn/helpers

Bumps [eslint](https://github.com/eslint/eslint) from 8.0.0 to 8.3.0.
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/main/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v8.0.0...v8.3.0)

---
updated-dependencies:
- dependency-name: eslint
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

jurre committed:
Merge pull request #4334 from phillipuniverse/poetry-file-dependency

Do not freeze file-based Poetry dependency version

jurre committed:
Merge pull request #4409 from AlekhyaYalla/optimize-yarn-updater

Run YarnUpdate only once for a version requirement

jurre committed:
Merge pull request #4414 from dependabot/dependabot/composer/composer/helpers/v2/friendsofphp/php-cs-fixer-3.3.2

build(deps-dev): bump friendsofphp/php-cs-fixer from 3.2.1 to 3.3.2 in /composer/helpers/v2

jurre committed:
Merge pull request #4415 from dependabot/dependabot/composer/composer/helpers/v1/friendsofphp/php-cs-fixer-2.19.3

build(deps-dev): bump friendsofphp/php-cs-fixer from 2.19.2 to 2.19.3 in /composer/helpers/v1

dependabot[bot] committed:
build(deps): bump pip from 21.2.4 to 21.3.1 in /python/helpers

Bumps [pip](https://github.com/pypa/pip) from 21.2.4 to 21.3.1.
- [Release notes](https://github.com/pypa/pip/releases)
- [Changelog](https://github.com/pypa/pip/blob/main/NEWS.rst)
- [Commits](https://github.com/pypa/pip/compare/21.2.4...21.3.1)

---
updated-dependencies:
- dependency-name: pip
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

jurre committed:
Remove test around updating unneeded transitive dependency

The behavior around this has changed in pipenv, but it does not seem
documented or even expected.

Essentially, this test was doing the following:

- add the transitive dependency to the Pipfile
- run `pipenv lock`
- check the version of the transitive dependency in the lockfile

Previous versions of pipenv would then remove the transitive dependency
from the lockfile, but it is not clear to me why, as it is present in
the Pipfile at that point. My guess is it would keep the top-level
dependencies cached somehow, but I've not been able to find much.

Either way, if transitive dependencies are _actually_ removed (meaning,
not also present in the manifest file), this should still work just
fine. The test setup is at the least confusing and at the worst wrong,
let's remove it.

mctofu committed:
Clarify how env vars are passed to shell commands

Co-Authored-By: Mattt Zmuda <mattt@github.com>
Co-Authored-By: Landon Grindheim <landongrindheim@github.com>

NicolasT committed:
Python: Honour `--strip-extras` flag of `pip-compile`

The `--strip-extras` flag of `pip-compile` is useful to generate
contstraints files (instead of the more traditional requirements files)
from some `constraints.in` file.

Constraints files can't have extras markers on the packages (this
wouldn't make much sense indeed), hence they need to be stripped. Since
`pip-tools` 6.2.0 this is supported by using the `--strip-extras` flag.

This commit adds this as a 'recognized' flag: if it's found in an
existing `foo.txt` file, it will be passed to `pip-compile` when some
`foo.in` file is updated.

See: https://github.com/dependabot/dependabot-core/pull/3974#issuecomment-866660246
See: https://github.com/dependabot/dependabot-core/pull/3974#issuecomment-975953752
See: https://github.com/jazzband/pip-tools/issues/1300#issuecomment-860020431

jurre committed:
Merge pull request #4408 from dependabot/dependabot/pip/python/helpers/pipenv-2021.11.15

build(deps): bump pipenv from 2021.5.29 to 2021.11.15 in /python/helpers

jurre committed:
Merge pull request #4439 from NicolasT/pip-compile-strip-extras

Python: Honour `--strip-extras` flag of `pip-compile`

lseppala committed:
Use redirect.github.com for redirect service

lseppala committed:
Merge pull request #4441 from dependabot/lsep/redirect-service

Use redirect.github.com for redirect service

jurre committed:
Merge pull request #4352 from AlekhyaYalla/toggle-azure-labels-creation

[Azure] Check & Raise TagsCreationForbidden Exception

lseppala committed:
v0.168.0

lseppala committed:
Merge pull request #4443 from dependabot/v0.168.0-release-notes

v0.168.0

rouge8 committed:
python: Update `GIT_DEPENDENCY_UNREACHABLE_REGEX` for pip 21.3.1

jurre committed:
Merge pull request #4444 from rouge8/pip-21.3.1

python: Upgrade to pip 21.3.1

mctofu committed:
Merge pull request #4434 from dependabot/mctofu/go-mod-list-versions

Use `go list -m -versions` to determine available versions of a go module

mctofu committed:
v0.169.0

jeffwidman committed:
Bump minimum to 1.17

This isn't strictly necessary, but since the rest of the infra bumped to 1.17,
might as well bump it here too. Esp since the `go.mod` behavior changed
a bit in `1.17`... so if more libs get added ever, this makes it so `go.mod`/`go.sum`
will follow the new behavior...

mctofu committed:
Merge pull request #4446 from dependabot/v0.169.0-release-notes

v0.169.0

lseppala committed:
Merge pull request #4447 from jeffwidman/patch-1

Bump minimum to 1.17

dependabot[bot] committed:
build(deps): bump pipenv in /python/helpers

Bumps [pipenv](https://github.com/pypa/pipenv) from 2021.11.15 to 2021.11.23.
- [Release notes](https://github.com/pypa/pipenv/releases)
- [Changelog](https://github.com/pypa/pipenv/blob/main/CHANGELOG.rst)
- [Commits](https://github.com/pypa/pipenv/compare/v2021.11.15...v2021.11.23)

---
updated-dependencies:
- dependency-name: pipenv
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

jurre committed:
Merge pull request #4452 from dependabot/dependabot/pip/python/helpers/pipenv-2021.11.23

build(deps): bump pipenv from 2021.11.15 to 2021.11.23 in /python/helpers

jurre committed:
Merge pull request #4436 from dependabot/dependabot/npm_and_yarn/npm_and_yarn/helpers/eslint-8.3.0

build(deps-dev): bump eslint from 8.0.0 to 8.3.0 in /npm_and_yarn/helpers

jurre committed:
Merge pull request #4430 from dependabot/dependabot/composer/composer/helpers/v1/phpstan/phpstan-1.2.0

build(deps-dev): bump phpstan/phpstan from 0.12.99 to 1.2.0 in /composer/helpers/v1

jurre committed:
Merge pull request #4431 from dependabot/dependabot/composer/composer/helpers/v2/phpstan/phpstan-1.2.0

build(deps-dev): bump phpstan/phpstan from 0.12.93 to 1.2.0 in /composer/helpers/v2

jurre committed:
Dependabot config: ignore npm lib major version updates

These helpers exist to support npm 6, so we require that specific major version and should not update it.

jurre committed:
Merge pull request #4454 from dependabot/jurre/ignore-npm-lib-major-updates

Dependabot config: ignore npm lib major version updates

jurre committed:
Update dependabot.yml

Commit:
Merge remote-tracking branch 'upstream/main' into test_failures
